src/shared_utilities/FileReader.jl

```diff
@@ -0,0 +1,371 @@
+
+Note that this conversion may fail if the date to convert doesn't
+exist in the DateTime calendar.
+
+# Arguments
+- `date`: object to be converted to DateTime
+"""
+to_datetime(date) = CFTime.reinterpret(DateTime, date)
+
+"""
+    datetime_to_strdate(datetime::DateTime)
+
+Convert from Date to String ("YYYYMMDD") format.
+
+# Arguments
+- `datetime`: [DateTime] object to be converted to string
+"""
+datetime_to_strdate(datetime::DateTime) =
+    string(lpad(Dates.year(datetime), 4, "0")) *
+    string(string(lpad(Dates.month(datetime), 2, "0"))) *
+    string(lpad(Dates.day(datetime), 2, "0"))
+
+"""
+    FileInfo
+
+Stores information specific to each boundary condition from a file and each variable.
+
+# Inputs:
+- file_path::S                    # directory of the data file
+- comms::X                        # communication context used for MPI
+- varname::V                      # name of the variable
+- all_dates::D                    # vector of all dates contained in the original data file
+- data_fields::C                  # tuple of two fields at consecutive dates, that will be used for interpolation
+- surface_space::B                # the space we're reading data onto
+- date_idx::Vector{Int}           # time index of the data in the file
+- date_idx0::Vector{Int}          # `date_idx` of the file data that is closest to date0
+- segment_length::Vector{Int}     # length of each date segment (used in the daily interpolation)
+- regrid_dirpath::S               # filename root for regridded data
+- outfile_root::S                 # string root for saved output files for this variable
+"""
+struct FileInfo{FT <: Real, S, X, V, D, C, B, VI}
+    file_path::S
+    comms::X
+    varname::V
+    all_dates::D
+    data_fields::C
+    surface_space::B
+    date_idx::VI
+    date_idx0::VI
+    segment_length::VI
+    regrid_dirpath::S
+    outfile_root::S
+end
+
+FileInfo{FT}(args...) where {FT} = FileInfo{FT, typeof.(args[1:7])...}(args...)
+float_type_fi(::FileInfo{FT}) where {FT} = FT
+
+
+"""
+    file_info_init(
+        FT,
+        regrid_dirpath,
+        file_path,
+        varname,
+        surface_space,
+        comms;
+        date_idx0 = nothing,
+        mono = true,
+    )
+
+Regrids from lat-lon grid to cgll grid, saving the output in a new file,
+and returns the info packaged in a single struct.
+
+Note: For now we assume our simulations always start at date0.
+This will need to be changed if we read in mulitple files over time that
+start at different dates, or if we want to run our simulation from a
+different date.
+We could address this by adding a `date_start` field,
+where the date at any time `t` in a simulation is given by `date_start + t`.
+This is relevant since the land model keeps track of time
+as the number of seconds elapsed since the starting time, rather
+than as a standalone DateTime.
+
+# Arguments
+- `FT`: [DataType] Float type.
+- `regrid_dirpath`: [String] directory the data file is stored in.
```

Comment:
```suggestion
- `regrid_dirpath`: String[] directory the data file is stored in.
```
I think it's better that we write this in a way that matches julia syntax

Reply:
The type is just String, not a String array. When we were refactoring the coupler we came up with this notation to denote argument types `- 'arg': [Type]`. If you think it's confusing though I can change it

Reply:
Ah, I see. I think we've actually done that in ClimaAtmos, too, now that I think about it. Part of me kind of likes the `::` notation since it follows julia syntax, it may be more familiar to users familiar with julia-- and since we can make arrays with brackets, it avoids the potential confusion with showing an example of types

Reply:
Yeah, that makes sense. I'm working on splitting this large struct into smaller structs with a more logical structure, so I can make this change as I go through as well.